src/component/Tooltip.tsx

```diff
@@ -90,6 +53,33 @@ export type TooltipProps<TValue extends ValueType, TName extends NameType> = Def
   useTranslate3d?: boolean;
 };
 
+function ContextUpdater(props: TooltipProps<any, any>) {
+  const [, setContext] = useContext(ChartContext);
+  useEffect(() => {
+    const newContext: ChartContextType = {
+      cursor: props.cursor,
+      isAnimationActive: props.isAnimationActive,
+      animationDuration: props.animationDuration,
+      animationEasing: props.animationEasing,
+      filterNull: props.filterNull,
+      // @ts-expect-error TODO this function will refine from the "outside, flexible prop" to "inside, strict prop" type
+      payloadUniqBy: props.payloadUniqBy,
+      wrapperStyle: props.wrapperStyle,
+      useTranslate3d: props.useTranslate3d,
+      // @ts-expect-error TODO
+      allowEscapeViewBox: props.allowEscapeViewBox,
+      // @ts-expect-error TODO
+      reverseDirection: props.reverseDirection,
+      offsetTopLeft: props.offset,
+      // @ts-expect-error TODO
+      position: props.position,
+    };
+    setContext(newContext);
```

Comment:
The public `Tooltip` will now only gather and cleanup the public props and push to context - nothing more.